lib/spack/spack/build_systems/cmake.py

```diff
@@ -32,6 +32,50 @@
 from spack.package import PackageBase, run_after
 
 
+def processCMakeCacheFile(cache_file):
+    def determineLineInterest(line):
+        return not line.startswith("//") \
+            and not line.startswith("#") \
+            and len(line) > 0 \
+            and not ":INTERNAL" in line \
+            and not ":PATH" in line \
+            and not ":FILEPATH" in line \
+            and not ":STATIC" in line \
+            and not "CMAKE_" in line \
+            and line.count("/") < 4  # *really* don't want to capture paths
```

Comment:
Filter this after parsing.  The `re.findall` (see below) gives you tuples of `(key, type, value)`.  Oh, and my regex already filtered `#` and `//`.  Then just modify what I wrote below:

```python
omit_types = ('INTERNAL', 'PATH', 'FILEPATH', 'STATIC')

with open(cache_file) as f:
    cache = dict((key, value) for key, type, value in
        re.findall(r'^([^#/:]*):(?:([^=]*)=)?([^\s]*)$', f.read(), re.M)
        if type not in omit_types and not value.count('/') > 4 and not key.startswith('CMAKE_'))
```